crates/sui-graphql-rpc/src/types/dot_move/named_move_package.rs

```diff
@@ -26,54 +33,96 @@ impl NamedMovePackage {
+            ResolutionType::External => {
+                query_external(ctx, config, versioned, checkpoint_viewed_at).await
+            }
+        }
     }
+}
 
-    async fn query_internal(
-        ctx: &Context<'_>,
-        config: &MoveRegistryConfig,
-        versioned: VersionedName,
-        checkpoint_viewed_at: u64,
-    ) -> Result<Option<MovePackage>, Error> {
-        let df_id = versioned.name.to_dynamic_field_id(config).map_err(|e| {
-            Error::Internal(format!("Failed to convert name to dynamic field id: {}", e))
-        })?;
+async fn query_internal(
+    ctx: &Context<'_>,
+    config: &MoveRegistryConfig,
+    versioned: VersionedName,
+    checkpoint_viewed_at: u64,
+) -> Result<Option<MovePackage>, Error> {
+    let df_id = versioned.name.to_dynamic_field_id(config).map_err(|e| {
+        Error::Internal(format!("Failed to convert name to dynamic field id: {}", e))
+    })?;
 
-        let Some(df) =
-            MoveObject::query(ctx, df_id.into(), Object::latest_at(checkpoint_viewed_at)).await?
-        else {
-            return Ok(None);
-        };
+    let Some(df) =
+        MoveObject::query(ctx, df_id.into(), Object::latest_at(checkpoint_viewed_at)).await?
+    else {
+        return Ok(None);
+    };
 
-        let app_record = AppRecord::try_from(df.native)?;
+    let app_record = AppRecord::try_from(df.native)?;
 
-        let Some(app_info) = app_record.app_info else {
-            return Ok(None);
-        };
+    let Some(app_info) = app_record.app_info else {
+        return Ok(None);
+    };
 
-        Self::package_from_app_info(ctx, app_info, versioned.version, checkpoint_viewed_at).await
-    }
+    package_from_app_info(ctx, app_info, versioned.version, checkpoint_viewed_at).await
+}
 
-    async fn package_from_app_info(
-        ctx: &Context<'_>,
-        app_info: AppInfo,
-        version: Option<u64>,
-        checkpoint_viewed_at: u64,
-    ) -> Result<Option<MovePackage>, Error> {
-        let Some(package_address) = app_info.package_address else {
-            return Ok(None);
-        };
-
-        // let's now find the package at a specified version (or latest)
-        MovePackage::query(
-            ctx,
-            package_address.into(),
-            if let Some(v) = version {
-                MovePackage::by_version(v, checkpoint_viewed_at)
-            } else {
-                MovePackage::latest_at(checkpoint_viewed_at)
-            },
-        )
-        .await
+async fn query_external(
+    ctx: &Context<'_>,
+    config: &MoveRegistryConfig,
+    versioned: VersionedName,
+    checkpoint_viewed_at: u64,
+) -> Result<Option<MovePackage>, Error> {
+    if config.external_api_url.is_none() {
+        return Err(MoveRegistryError::ExternalApiUrlUnavailable.into());
     }
+
+    let identifier: ChainIdentifier = *ctx.data_unchecked();
+
+    let Some(chain_id) = identifier.0 else {
+        return Err(MoveRegistryError::ChainIdentifierUnavailable.into());
+    };
```

Comment:
nit: Combine this into one statement
```suggestion
    let ChainIdentifier(Some(&chain_id)) = ctx.data_unchecked() else {
        return Err(MoveRegistryError::ChainIdentifierUnavailable.into());
    };
```